src/webapi/html_elements/image.rs

```diff
@@ -0,0 +1,293 @@
+    pub fn complete( &self ) -> bool {
+        js! (
+            return @{self}.complete;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the Cross-Origin Resource Sharing (CORS) setting for the image.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/crossOrigin)
+    pub fn cross_origin( &self ) -> CrossOriginSetting {
+        match js!(
+            return @{self}.crossOrigin;
+        ) {
+            Value::Null => CrossOriginSetting::None,
+            Value::String( ref s ) if *s == "anonymous" => CrossOriginSetting::Anonymous,
+            Value::String( ref s ) if *s == "use-credentials" => CrossOriginSetting::UseCredentials,
+            _ => unreachable!("Unexpected crossOrigin value")
+        }
+    }
+
+    /// Sets the Cross-Origin Resource Sharing (CORS) setting for the image.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/crossOrigin)
+    pub fn set_cross_origin( &self, value: CrossOriginSetting ) {
+        js! { @(no_return)
+            @{self}.crossOrigin = @{
+                match value {
+                    CrossOriginSetting::None => None,
+                    CrossOriginSetting::Anonymous => Some("anonymous"),
+                    CrossOriginSetting::UseCredentials => Some("use-credentials")
+                }
+            }
+        }
+    }
+
+    /// Returns the the rendered height of the image in CSS pixels.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/height)
+    pub fn height( &self ) -> u32 {
+        js! (
+            return @{self}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Sets the the rendered height of the image in CSS pixels.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/height)
+    pub fn set_height( &self, value: u32 ) {
+        js! { @(no_return)
+            @{self}.height = @{value};
+        }
+    }
+
+    /// Indicates whether the image is part of a server-side image map.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/isMap)
+    pub fn is_map( &self ) -> bool {
+        js!(
+            return @{self}.isMap;
+        ).try_into().unwrap()
+    }
+
+    /// Sets whether the image is part of a server-side image map.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/isMap)
+    pub fn set_is_map( &self, value: bool ) {
+        js! { @(no_return)
+            @{self}.isMap = @{value};
+        }
+    }
+
+    /// Returns the intrinsic height of the image in CSS pixels, if it is available;
+    /// else, it returns 0.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/naturalHeight)
+    pub fn natural_height( &self ) -> u32 {
```

Comment:
It would probably be better to return an `Option< u32 >` here, and in case of a zero return `None`. (As far I see from the documentation `0` means the lack of a value, which screams `Option` to me)